examples/data-objects/inventory-app/src/legacySharedTreeInventoryList.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	Change,
+	ChangeNode,
+	Definition,
+	SharedTree as LegacySharedTree,
+	SharedTreeEvent,
+	StablePlace,
+	TraitLabel,
+} from "@fluid-experimental/tree";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+
+import { IInventoryListUntyped, IPart } from "./interfaces";
+
+const legacySharedTreeKey = "legacySharedTree";
+
+/**
+ * An inventory list implemented using LegacySharedTree.
+ */
+export class LegacySharedTreeInventoryList extends DataObject implements IInventoryListUntyped {
+	private _tree: LegacySharedTree | undefined;
+	private get tree() {
+		if (this._tree === undefined) {
+			throw new Error("Not initialized properly");
+		}
+		return this._tree;
+	}
+
+	protected async initializingFirstTime() {
+		const legacySharedTree = this.runtime.createChannel(
+			undefined,
+			LegacySharedTree.getFactory().type,
+		) as LegacySharedTree;
+
+		// Initialize the inventory with two parts at zero quantity
+		// REV: Is ChangeNode appropriate here? Or should I use some other type (BuildNode?)?
+		const inventoryNode: ChangeNode = {
+			identifier: legacySharedTree.generateNodeId(),
+			// REV: This is based on the Definition casting from Bubblebench, it doesn't really seem
+			// to matter what is used here though.  Is there a better way to omit the definition here?
+			definition: "array" as Definition,
```

Comment:
this is domain-defined. If the domain you're modeling is an inventory which has some nuts and bolts, the sensible definition for this node would be "Inventory".

Since legacy shared-tree doesn't have a built-in schema system, I think probably all the definitions you use are somewhat arbitrary--might not even matter if you used `definition: "any"` everywhere (though it'd make debugging more confusing)